### PR TITLE
Update ride flags 7, 16, 18

### PR DESCRIPTION
**Pull request by spacek531**
Objects repository companion PR: ~~https://github.com/OpenRCT2/objects/pull/21~~ https://github.com/OpenRCT2/objects/pull/20

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
In https://github.com/OpenRCT2/objects/pull/20, @Gymnasiast opted for 'no' rather than 'disabled'. I'm inclined to go with 'disabled' myself, too. Even though flags are binary, some might read 'no' as shorthand for 'number'.

**Comment by IntelOrca** [comment]
I tried to avoid negatives with all the other flags.

**Comment by Gymnasiast** [comment]
There are a lot of flags that have negative names. I don't think it's really worth it to invert them.

**Comment by spacek531** [comment]
I chose the word "disable" for consistency, because several other flags use that word with the json files.

**Comment by spacek531** [comment]
@duncanspumpkin I think I discovered the purpose of `VEHICLE_ENTRY_FLAG_15`, so it will need updating in the objects repo as well.

File: src/openrct2/object/RideObject.cpp [hunk 1]
```diff
@@ -831,7 +831,7 @@ rct_ride_entry_vehicle RideObject::ReadJsonCar(const json_t * jCar)
         { "VEHICLE_ENTRY_FLAG_OVERRIDE_NUM_VERTICAL_FRAMES", VEHICLE_ENTRY_FLAG_OVERRIDE_NUM_VERTICAL_FRAMES },
         { "VEHICLE_ENTRY_FLAG_13", VEHICLE_ENTRY_FLAG_13 },
         { "VEHICLE_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES", VEHICLE_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES },
-        { "VEHICLE_ENTRY_FLAG_15", VEHICLE_ENTRY_FLAG_15 },
+        { "VEHICLE_ENTRY_FLAG_LIFT", VEHICLE_ENTRY_FLAG_LIFT },
```
Comment (Broxzier) [inline src/openrct2/object/RideObject.cpp]: What does this mean? I assume it's not the same as the lift ride type.

Comment (Gymnasiast) [reply src/openrct2/object/RideObject.cpp]: It is. It is only set on the lift cabin, mine lift cabin, and blob-from-outer-space lift cabin.

(Spacek also explained this in the issue)

Comment (Broxzier) [reply src/openrct2/object/RideObject.cpp]: Ah, I didn't see the edit. At first I was thinking it was meant for rides that have a steep hill lift or something, like the giga coaster. This flag seems pretty pointless then, if the ride type can be checked. Same goes for the mini golf flag earlier on in the list.

Comment (spacek531) [reply src/openrct2/object/RideObject.cpp]: I tested the observation tower on a lift ride and it failed to move. I think the flag is necessary for the lift to work.

**Comment by Gymnasiast** [comment]
Ok, my plan for all the flags:

* `RIDE_ENTRY_FLAG_7` -> remove altogether
* `RIDE_ENTRY_FLAG_16` -> move to ride groups, as it affects building
* `RIDE_ENTRY_FLAG_18` -> rename
* `VEHICLE_ENTRY_FLAG_15` -> rename to `VEHICLE_ENTRY_FLAG_LIFT`

**Comment by Gymnasiast** [comment]
I have done the above things locally, but it still needs an object release.

File: src/openrct2/ride/RideGroupManager.h [hunk 2]
```diff
@@ -40,3 +41,8 @@ class RideGroupManager
     static int32_t VehiclePreferenceCompare(const uint8_t rideType, const char * a, const char * b);
     static bool RideTypeIsIndependent(const uint8_t rideType);
 };
+
+enum RideGroupFlags : uint8_t
+{
+    RIDE_GROUP_FLAG_ALLOW_DOORS_ON_TRACK = 0b00000001,
+};
```
Comment (Broxzier) [inline src/openrct2/ride/RideGroupManager.h]: Curious, which editor do you use? `.editorconfig` has a rule that adds a new line before the end of file.

**Pull request merged**